packages/calcite-components/src/utils/dom.ts

```diff
@@ -565,12 +576,25 @@ export function slotChangeHasAssignedElement(event: Event): boolean {
  * ```
  *
  * @param {Event} event The event.
- * @returns {boolean} Whether the slot has any assigned elements.
+ * @param {string} selector The CSS selector string to filter the returned elements by.
+ * @returns {Element[]} An array of elements.
+ */
+export function slotChangeGetAssignedElements(event: Event, selector?: string): Element[] {
+  return getSlotAssignedElements(event.target as HTMLSlotElement, selector);
+}
+
+/**
+ * This helper returns the assigned elements on a `slot` element, filtered by an optional css selector.
+ *
+ * @param {HTMLSlotElement} slot The slot element.
+ * @param {string} selector CSS selector string to filter the returned elements by.
+ * @returns {Element[]} An array of elements.
  */
-export function slotChangeGetAssignedElements(event: Event): Element[] {
-  return (event.currentTarget as HTMLSlotElement).assignedElements({
+export function getSlotAssignedElements<T>(slot: HTMLSlotElement, selector?: string): T[] {
+  const assignedElements = slot.assignedElements({
     flatten: true,
   });
+  return selector ? (filterElementsByCSSSelector(assignedElements, selector) as T[]) : (assignedElements as T[]);
```

Comment:
I _think_ you can avoid all typecasting if you pass T (after the generic type is updated) as the type to supporting functions:

```tsx
return selector ? 
  filterElementsByCSSSelector<T>(assignedElements, selector) : 
  assignedElements;
```

Reply:
Works for the `filter*` function call, but not for the assignedElements variable:

![image](https://github.com/user-attachments/assets/f61edd39-8629-44fd-a0ba-4be59fb65f17)

![image](https://github.com/user-attachments/assets/6994e605-eef2-4572-bdab-df266d42ead7)